libs/telemetry/src/test/java/org/opensearch/telemetry/tracing/DefaultTracerTests.java

```diff
@@ -249,75 +250,66 @@ public void testEndSpanByClosingSpanScopeMultiple() {
      * 4. verify the current_span is still the same on async thread as the 2
      * 5. verify the main thread has current span as null.
      */
-    public void testSpanAcrossThreads() {
+    public void testSpanAcrossThreads() throws ExecutionException, InterruptedException {
         TracingTelemetry tracingTelemetry = new MockTracingTelemetry();
         ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
-        AtomicReference<Span> currentSpanRefThread1 = new AtomicReference<>();
-        AtomicReference<Span> currentSpanRefThread2 = new AtomicReference<>();
-        AtomicReference<Span> currentSpanRefAfterEndThread2 = new AtomicReference<>();
-
-        AtomicReference<Span> spanRef = new AtomicReference<>();
-        AtomicReference<Span> spanT2Ref = new AtomicReference<>();
 
         ThreadContextBasedTracerContextStorage spanTracerStorage = new ThreadContextBasedTracerContextStorage(
             threadContext,
             tracingTelemetry
         );
         DefaultTracer defaultTracer = new DefaultTracer(tracingTelemetry, spanTracerStorage);
 
-        executorService.execute(() -> {
+        CompletableFuture<?> asyncTask = CompletableFuture.runAsync(() -> {
             // create a span
             Span span = defaultTracer.startSpan(new SpanCreationContext("span_name_t_1", Attributes.EMPTY));
             SpanScope spanScope = defaultTracer.withSpanInScope(span);
-            spanRef.set(span);
 
-            executorService.execute(() -> {
+            CompletableFuture<?> asyncTask1 = CompletableFuture.runAsync(() -> {
                 Span spanT2 = defaultTracer.startSpan(new SpanCreationContext("span_name_t_2", Attributes.EMPTY));
                 SpanScope spanScopeT2 = defaultTracer.withSpanInScope(spanT2);
-                spanT2Ref.set(spanT2);
-
-                currentSpanRefThread2.set(defaultTracer.getCurrentSpan().getSpan());
+                assertEquals(spanT2, defaultTracer.getCurrentSpan().getSpan());
 
-                spanT2.endSpan();
                 spanScopeT2.close();
-                currentSpanRefAfterEndThread2.set(getCurrentSpanFromContext(defaultTracer));
-            });
+                spanT2.endSpan();
+                assertEquals(null, defaultTracer.getCurrentSpan());
+            }, executorService);
+            asyncTask1.join();
             spanScope.close();
-            currentSpanRefThread1.set(getCurrentSpanFromContext(defaultTracer));
-        });
-        assertEquals(spanT2Ref.get(), currentSpanRefThread2.get());
-        assertEquals(spanRef.get(), currentSpanRefAfterEndThread2.get());
-        assertEquals(null, currentSpanRefThread1.get());
+            span.endSpan();
+            assertEquals(null, defaultTracer.getCurrentSpan());
+        }, executorService);
+        asyncTask.get();
```

Comment:
Please use `join` as well:

```suggestion
        asyncTask.join();
```